athena-vertica/athena-vertica.yaml

```diff
@@ -0,0 +1,102 @@
+# to your template when you create a stack
+Parameters:
+  AthenaCatalogName:
+    Description: 'The name you will give to this catalog in Athena. It will also be used as the function name. This name must satisfy the pattern ^[a-z0-9-_]{1,64}$'
+    Type: String
+    AllowedPattern: ^[a-z0-9-_]{1,64}$
+    Default: "v223"
+  SpillBucket:
+    Description: 'The name of the bucket where this function can spill data.'
+    Type: String
+    Default: "test-connector-west"
+  DataBucket:
+    Description: "The bucket where this tutorial's data lives."
+    Type: String
+    Default: "test-connector-west"
+  ExportBucket:
+    Description: "The bucket where the Vertica Query results will be exported."
+    Type: String
+    Default: "test-connector-west"
+  SpillPrefix:
+    Description: 'The prefix within SpillBucket where this function can spill data.'
+    Type: String
+    Default: athena-spill
+  LambdaTimeout:
+    Description: "Maximum Lambda invocation runtime in seconds. (min 1 - 900 max)"
+    Default: 900
+    Type: Number
+  LambdaMemory:
+    Description: "Lambda memory in MB (min 128 - 3008 max)."
+    Default: 3008
+    Type: Number
+  DisableSpillEncryption:
+    Description: "WARNING: If set to 'true' encryption for spilled data is disabled."
+    Default: "false"
+    Type: String
+  VerticaConnectionString:
+    Description: 'The Vertica connection details to use by default if not catalog specific connection is defined and optionally using SecretsManager (e.g. ${secret_name}).'
+    Type: String
+    Default: "jdbc:vertica://<HOST_NAME>:<PORT_NAME>/<DB_NAME>?user=<USER>&password=<PASSWORD>"
+  AWSSecretsManagerSecretName:
+    Description: "Secret Name of AWS Access Keys in AWS Secrets Manager"
+    Type: String
+    Default: "MyAwsCreds"
+  AWSRegion:
+    Description: "Region Name"
+    Type: String
+    Default: "US-WEST-2"
+
+Resources:
+  ConnectorConfig:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      Environment:
+        Variables:
+          disable_spill_encryption: !Ref DisableSpillEncryption
+          spill_bucket: !Ref SpillBucket
+          spill_prefix: !Ref SpillPrefix
+          data_bucket: !Ref DataBucket
+          export_bucket: !Ref ExportBucket
+          default_vertica: !Ref VerticaConnectionString
+          aws_secret_name: !Ref AWSSecretsManagerSecretName
+          aws_region: !Ref AWSRegion
+
+      FunctionName: !Sub "${AthenaCatalogName}"
+      Handler: "com.amazonaws.connectors.athena.vertica.VerticaCompositeHandler"
+      CodeUri: "./target/athena-vertica-1.0.jar"
+      Description: "A guided example for writing and deploying your own federated Amazon Athena connector for a custom source."
+      Runtime: java8
+      Timeout: !Ref LambdaTimeout
+      MemorySize: !Ref LambdaMemory
+      Policies:
+        - Statement:
+            - Action:
+                - athena:GetQueryExecution
+                - s3:ListAllMyBuckets
+                - lambda:InvokeFunction
```

Comment:
shouldnt need this anymore

I also don't see access to read S3 objects that vertica writes, where is that handled?

Reply:
Removed the Lambda Invoke permission and added S3 permissions